app/assets/javascripts/host_edit.js

```diff
@@ -296,7 +255,6 @@ function handleHostgroupChangeEdit(element, host_id, host_changed) {
   } else {
     // edit host
     set_inherited_value(element);
-    update_puppetclasses(element);
```

Comment:
How do we handle this and other events that should reload/update one of the puppet tabs?

Reply:
I've moved all these functions to the webpack file in plugin and added the events to the global js file:
https://github.com/theforeman/foreman_puppet_enc/pull/64/files#diff-c26a70779db93ece76946fdfe9f914a20986c4177863883f79a75dab2d7bd90eR14-R30

I've tried hard to decode when should the tabs/info get reloaded and do the same.
One difference that I didn't know how to avoid is reloading the puppet parameters on OS and domain change. Now we are changing it after the ajax (`os_changed`, `domain_change`) success, but in the plugin I'm running them right away after the selection. Hopefully it should not be an issue and I've tested it's ok, but I'm not sure if I've missed something that could get affected by that :thinking: